### ISSUE TITLE
migration from python2 to python3

### ISSUE THREAD

**Issue by tcezard** [SciLifeLab/genologics#132]
Hi I just stumbled upon your clarity REST client interface and thought it could be very useful.
However all our code is in python3 
I was wondering if there was any interest in using a python3 version if we make the migration effort

Cheers
Tim

**Comment by Galithil** [comment]
That is an interesting question that I'll bring up at our next developer's meeting on Monday.

Currently, all of our infrastructure uses Python2, so the odds are that a python 3 version would not be of much use to us. I'll keep you updated with the discussion.

**Comment by tcezard** [comment]
Cool thank you.
It is possible to run python3 code that run under python2 provided you use python 2.7 and we import the proper __future__ package where necessary.
If it doesn't work I'm happy to keep pulling from your master to keep things in sync but I am less likely to contribute anything back.

**Comment by Galithil** [comment]
Most of the code in this repo was written for python 2.6. While it runs
fine under 2.7, I wouldn't make any assumptions as to wether it would work
on Python3.

**Comment by tcezard** [comment]
I've already done a migration yesterday and with a bit of work got it to work fine under 3.4. Now there would be more work needed to make sure the code I have could run under 2.6/2.7 but it is possible. The problem would probably be with libraries that have move like "urljoin" but nothing we couldn't work out.
Wether you're interested or not I want to thank you for making your code open source and MIT licensed. It saves us a great deal of work.

**Comment by Galithil** [comment]
You're welcome.
The other developers did not see any issues with making the code python3 compatible, as long as it stays compatible with 2.7. You're welcome to make a PR, and we'll ensure that it does not impact our systems too much.

**Comment by jwhite007** [comment]
I've been able to port the package over to python3 with no issues. I used 2to3 and then had to make minor adjustments to functions such as urljoin as you mention, tcezard. Everything is working fine.

**Comment by tcezard** [comment]
Sorry I've left this silent for so long.
I'm working on unit testing and travis CI integration at the moment. 
once that is done I'll do a pull request.
Doing travis is important as it will help maintaining functionality for both python 2.7 and 3+
I'm working [here](https://github.com/EdinburghGenomics/genologics/blob/test_lims/tests/test_lims.py) if you want to see the progress.
